Stage II/Single_Year_Stage_II.py

```diff
@@ -13,6 +13,10 @@ def Single_Year_Stage_II(puf, Stage_I_factors, Stage_II_targets, year, tol):
 
     print("Preparing coefficient matrix...")
 
+    # the first half of this function (with all of the np.where statements)
+    # might be amenable to refactoring to a numba function, could improve
+    # readability
```

Comment:
I think that's a good idea, especially if everything has the same dimensionality.